src/main/java/zowe/client/sdk/zosjobs/SubmitJobs.java

```diff
@@ -194,15 +199,17 @@ public Job submitJobCommon(SubmitJobParams params) throws Exception {
         var jsonRequestBody = new JSONObject(jsonMap);
         LOG.debug(String.valueOf(jsonRequestBody));
 
-        if (params.getJclSymbols().isPresent()) {
-            // TODO..
-        }
-
         if (request == null || !(request instanceof JsonPutRequest)) {
             request = ZoweRequestFactory.buildRequest(connection, ZoweRequestType.VerbType.PUT_JSON);
         }
+
         request.setRequest(url, jsonRequestBody.toString());
 
+        if (params.getJclSymbols().isPresent()) {
+            Map<String, String> extraHeaders = getSubstitutionHeaders(params.getJclSymbols().get());
+            request.setHeaders(extraHeaders);
+        }
+
         Response response = request.executeRequest();
```

Comment:
you may be able to replace this entire if statement with the following:
params.getJclSymbols().ifPresent(syms -> request.setHeaders(getSubstitutionHeaders(syms));

you a suggest play with some new magic.  not required. otherwise looks well!